### ISSUE TITLE
core-agent-manager script crashes

### ISSUE THREAD

**Issue by aaugustin** [scoutapp/scout_apm_python#140]
Scout installs a `core-agent-manager` script through a setuptools entry point.

This script currently appears to be broken:

```
myk@mYk:~ $ mkvirtualenv test_scout_apm
WARNING: the pyenv script is deprecated in favour of `python3.7 -m venv`
virtualenvwrapper.user_scripts creating /Users/myk/.virtualenvs/test_scout_apm/bin/predeactivate
virtualenvwrapper.user_scripts creating /Users/myk/.virtualenvs/test_scout_apm/bin/postdeactivate
virtualenvwrapper.user_scripts creating /Users/myk/.virtualenvs/test_scout_apm/bin/preactivate
virtualenvwrapper.user_scripts creating /Users/myk/.virtualenvs/test_scout_apm/bin/postactivate
virtualenvwrapper.user_scripts creating /Users/myk/.virtualenvs/test_scout_apm/bin/get_env_details

(test_scout_apm) myk@mYk:~ $ pip install scout-apm
Collecting scout-apm
  Downloading https://files.pythonhosted.org/packages/54/45/762ae1ff59d8dc5209d8c9f0b8f30c19dd48fda90eb9822e8fcfd12cf145/scout-apm-1.3.8.tar.gz (41kB)
    100% |████████████████████████████████| 51kB 1.9MB/s
Collecting psutil (from scout-apm)
Collecting PyYAML (from scout-apm)
Collecting requests (from scout-apm)
  Using cached https://files.pythonhosted.org/packages/ff/17/5cbb026005115301a8fb2f9b0e3e8d32313142fe8b617070e7baad20554f/requests-2.20.1-py2.py3-none-any.whl
Collecting urllib3<1.25,>=1.21.1 (from requests->scout-apm)
  Using cached https://files.pythonhosted.org/packages/62/00/ee1d7de624db8ba7090d1226aebefab96a2c71cd5cfa7629d6ad3f61b79e/urllib3-1.24.1-py2.py3-none-any.whl
Collecting idna<2.8,>=2.5 (from requests->scout-apm)
  Using cached https://files.pythonhosted.org/packages/4b/2a/0276479a4b3caeb8a8c1af2f8e4355746a97fab05a372e4a2c6a6b876165/idna-2.7-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests->scout-apm)
  Downloading https://files.pythonhosted.org/packages/9f/e0/accfc1b56b57e9750eba272e24c4dddeac86852c2bebd1236674d7887e8a/certifi-2018.11.29-py2.py3-none-any.whl (154kB)
    100% |████████████████████████████████| 163kB 4.5MB/s
Collecting chardet<3.1.0,>=3.0.2 (from requests->scout-apm)
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Installing collected packages: psutil, PyYAML, urllib3, idna, certifi, chardet, requests, scout-apm
  Running setup.py install for scout-apm ... done
Successfully installed PyYAML-3.13 certifi-2018.11.29 chardet-3.0.4 idna-2.7 psutil-5.4.8 requests-2.20.1 scout-apm-1.3.8 urllib3-1.24.1

(test_scout_apm) myk@mYk:~ $ $VIRTUAL_ENV/bin/core-agent-manager launch
Traceback (most recent call last):
  File "/Users/myk/.virtualenvs/test_scout_apm/bin/core-agent-manager", line 11, in <module>
    load_entry_point('scout-apm==1.3.8', 'console_scripts', 'core-agent-manager')()
  File "/Users/myk/.virtualenvs/test_scout_apm/lib/python3.7/site-packages/scout_apm/core/cli/core_agent_manager.py", line 39, in main
    globals()[kwargs.pop('subparser')](**kwargs)
  File "/Users/myk/.virtualenvs/test_scout_apm/lib/python3.7/site-packages/scout_apm/core/cli/core_agent_manager.py", line 15, in launch
    core_agent_manager = CoreAgentManager()
  File "/Users/myk/.virtualenvs/test_scout_apm/lib/python3.7/site-packages/scout_apm/core/core_agent_manager.py", line 26, in __init__
    self.core_agent_dir = '{}/{}'.format(AgentContext.instance.config.value('core_agent_dir'),
AttributeError: 'NoneType' object has no attribute 'config'
```

The error happens because `CoreAgentManager` attempts to access the `AgentContext` instance (which is intended to be a singleton) without callling `AgentContext.build()` first.

**Comment by adamchainz** [comment]
I've got a fix in #202, but since it has been broken for about a year with no complaints, we're discussing on Slack whether to remove it.

**Comment by adamchainz** [comment]
Fix was released in 2.2.0 - still undocumented but useful for the occasional customer support.
